code/modules/prototypes/prototype.dm

```diff
@@ -0,0 +1,875 @@
+			M.Turn(-90)
+			I.transform = M
+		if("E")
+			user << "<span class='notice'>The [owner] points east.</span>"
+			M.Turn(90)
+			I.transform = M
+	flick_overlay(I,list(user.client),50)
+	return
+
+/obj/effect/rift
+	name = "tear in the fabric of reality"
+	desc = "You should run now."
+	icon = 'icons/obj/biomass.dmi'
+	icon_state = "rift"
+	density = 1
+	unacidable = 1
+	anchored = 1.0
+	var/list/turf_paths = list()
+	var/list/mob_paths = list()
+	var/mobs_per_wave = 2
+	var/current_wave = 0
+	var/max_waves = 7
+	var/wave_delay = 450
+	var/wave_time = 0
+	var/obj/item/device/assembly/signaler/rift/aSignal = null
+
+/obj/effect/rift/attackby(obj/item/W,mob/living/user, params)
+	if(istype(W,/obj/item/weapon/nullrod))
+		for(var/obj/item/I in src)
+			I.loc = src.loc
+		user << "<span class='notice'>You disperse the [src] with the power of the [W]!</span>"
+		Neutralize()
+		
+	if(istype(W, /obj/item/device/analyzer))
+		user << "<span class='notice'>Analyzing... [src]'s unstable field is fluctuating along frequency [aSignal.code]:[format_frequency(aSignal.frequency)].</span>"
+
+/obj/effect/rift/proc/wave(wave)
+	current_wave++
+	if(current_wave == max_waves)
+		SSobj.processing.Remove(src)
+		return
+	for(var/turf/T in circlerange(src,wave))
+		T.ChangeTurf(pick(turf_paths))
+	for(var/i=0,i<mobs_per_wave,i++)
+		var/mob_p = pick(mob_paths)
+		new mob_p(get_turf(src))
+
+/obj/effect/rift/proc/Neutralize()
+	visible_message("<span class='warning'>[src] collapses!</span>")
+	SSobj.processing.Remove(src)
+	qdel(src)
+
+/obj/effect/rift/Destroy()
+	qdel(aSignal)
+	..()
+
+/obj/effect/rift/New()
+	wave_time = world.time + wave_delay
+	SSobj.processing |= src
+
+	aSignal = new(src)
+	aSignal.code = rand(1,100)
+
+	aSignal.frequency = rand(1200, 1599)
+	if(IsMultiple(aSignal.frequency, 2))//signaller frequencies are always uneven!
+		aSignal.frequency++
+
+/obj/effect/rift/process()
+	if(world.time >= wave_time)
+		wave(current_wave)
+		wave_time = world.time + wave_delay
+
+/obj/item/device/assembly/signaler/rift
+	name = "rift signaler"
+	desc = "You probably shouldn't see this"
+
+/obj/item/device/assembly/signaler/rift/receive_signal(datum/signal/signal)
+	if(!signal)
+		return 0
+	if(signal.encryption != code)
+		return 0
+	var/obj/effect/rift/R = src.loc
+	if(!istype(R))
+		qdel(src)
+	else
+		R.Neutralize()
+
+/obj/effect/rift/cult
+	turf_paths = list(/turf/simulated/floor/engine/cult)
+	mob_paths =  list(/mob/living/simple_animal/hostile/faithless) //replace with constructs when they get turned into /hostile
+	mobs_per_wave = 1
+/obj/effect/rift/hivebot
+	turf_paths = list(/turf/simulated/floor/bluegrid,/turf/simulated/floor/greengrid)
+	mob_paths =  list(/mob/living/simple_animal/hostile/hivebot,/mob/living/simple_animal/hostile/hivebot/range,/mob/living/simple_animal/hostile/hivebot/strong)
+	mobs_per_wave = 3
+
+/obj/effect/rift/alien
+	turf_paths = list(/turf/simulated/floor/plating/asteroid)
```

Comment:
Yeah, but there's no turf equivalent for these. I suppose i could just drop some nodes around